beginner_source/maskedtensor_sparsity.rst

```diff
@@ -0,0 +1,218 @@
+stands for "coordinate format", where the specified elements are stored as tuples of their indices and the
+corresponding values. That is, the following are provided:
+
+* ``indices``: array of size ``(ndim, nse)`` and dtype ``torch.int64``
+* ``values``: array of size `(nse,)` with any integer or floating point dtype
+
+where ``ndim`` is the dimensionality of the tensor and ``nse`` is the number of specified elements
+
+For both sparse COO and CSR tensors, you can construct a :class:`MaskedTensor` by doing either:
+
+1. ``masked_tensor(sparse_tensor_data, sparse_tensor_mask)``
+2. ``dense_masked_tensor.to_sparse_coo()`` or ``dense_masked_tensor.to_sparse_csr()``
+
+The second method is easier to illustrate so we've shown that below, but for more on the first and the nuances behind
+the approach, please read the :ref:`sparse-coo-appendix`.
+
+    >>> values = torch.tensor([[0, 0, 3], [4, 0, 5]])
+    >>> mask = torch.tensor([[False, False, True], [False, False, True]])
+    >>> mt = masked_tensor(values, mask)
+    >>> sparse_coo_mt = mt.to_sparse_coo()
+    >>> mt
+    MaskedTensor(
+      [
+        [      --,       --, 3],
+        [      --,       --, 5]
+      ]
+    )
+    >>> sparse_coo_mt
+    MaskedTensor(
+      [
+        [      --,       --, 3],
+        [      --,       --, 5]
+      ]
+    )
+    >>> sparse_coo_mt.get_data()
+    tensor(indices=tensor([[0, 1],
+                          [2, 2]]),
+          values=tensor([3, 5]),
+          size=(2, 3), nnz=2, layout=torch.sparse_coo)
+
+Sparse CSR tensors
+------------------
+
+Similarly, :class:`MaskedTensor` also supports the
+`CSR (Compressed Sparse Row) <https://pytorch.org/docs/stable/sparse.html#sparse-csr-tensor>`__
+sparse tensor format. Instead of storing the tuples of the indices like sparse COO tensors, sparse CSR tensors
+aim to decrease the memory requirements by storing compressed row indices.
+In particular, a CSR sparse tensor consists of three 1-D tensors:
+
+* ``crow_indices``: array of compressed row indices with size ``(size[0] + 1,)``. This array indicates which row
+  a given entry in values lives in. The last element is the number of specified elements,
+  while crow_indices[i+1] - crow_indices[i] indicates the number of specified elements in row i.
+* ``col_indices``: array of size ``(nnz,)``. Indicates the column indices for each value.
+* ``values``: array of size ``(nnz,)``. Contains the values of the CSR tensor.
+
+Of note, both sparse COO and CSR tensors are in a `beta <https://pytorch.org/docs/stable/index.html>`__ state.
+
+By way of example:
+
+    >>> mt_sparse_csr = mt.to_sparse_csr()
+    >>> mt_sparse_csr
+    MaskedTensor(
+      [
+        [      --,       --, 3],
+        [      --,       --, 5]
+      ]
+    )
+    >>> mt_sparse_csr.get_data()
+    tensor(crow_indices=tensor([0, 1, 2]),
+          col_indices=tensor([2, 2]),
+          values=tensor([3, 5]), size=(2, 3), nnz=2, layout=torch.sparse_csr)
+
+Appendix
```

Comment:
nit: This is definitely a personal preference but I think that this tutorial is not long enough to warrant an appendix. As a reader, I would take "appendix" as an invitation to stop reading but the part on how data and mask could have different nnz and data will assume the shape of mask seems like an important (and potentially confusing) detail.

Personally, would delete the header "appendix" and just keep the rest of it as is :b